src/common/__tests__/merjoonTransformer.test.ts

```diff
@@ -65,9 +65,69 @@ describe('MerjoonTransformer', () => {
         expect(key).toBe('string("content")');
       });
     });
+
+    describe('matches', () => {
+      it('match is not null', () => {
+        const { type, key} = MerjoonTransformer.parseTypedKey('UUID("remote_id")');
+
+        expect(type).toBe('UUID');
+        expect(key).toBe('remote_id');
+      });
+    });
+
+    describe('does not match', () => {
+      it('match is null', () => {
+        const { type, key} = MerjoonTransformer.parseTypedKey('remote_id');
+
+        expect(type).toBeUndefined();
+        expect(key).toBe('remote_id');
+      });
+    });
   });
 
   describe('parseValue', () => {
+    describe('UUID', () => {
+      it('should assign hashed value to newVal', () => {
+        const data = {
+          accountId: '712020:950855f3-65cc-4b69-b797-0f2f60973fd1',
+        };
+        const path = 'UUID("accountId")';
+        const value = MerjoonTransformer.parseValue(data, path);
+
+        expect(value).toBe('58e957f4607f014a3bf04664a7f0eb6f');
+      });
+    });
+
+    describe('STRING', () => {
+      it('should assign string value to newVal', () => {
```

Comment:
rename this to `should return string if parsing value is number` and add one more `should return string if parsing value is string`